service/javadsl/api/src/main/scala/com/lightbend/lagom/internal/javadsl/api/ServiceReader.scala

```diff
@@ -28,13 +30,28 @@ object ServiceReader {
 
   final val DescriptorMethodName = "descriptor"
 
+  private val isJava8 = java.lang.management.ManagementFactory.getRuntimeMXBean.getSpecVersion.startsWith("1.8")
+
   /**
    * In order to invoke default methods of a proxy in Java 8 reflectively, we need to create a MethodHandles.Lookup
    * instance that is allowed to lookup private methods. The only way to do that is via reflection.
    */
-  private val methodHandlesLookupConstructor = classOf[MethodHandles.Lookup].getDeclaredConstructor(classOf[Class[_]], classOf[Int])
-  if (!methodHandlesLookupConstructor.isAccessible) {
-    methodHandlesLookupConstructor.setAccessible(true)
+  private val defaultMethodHandleSupplier: Method => MethodHandle = {
+    if (isJava8) {
+      val methodHandlesLookupConstructor = classOf[MethodHandles.Lookup].getDeclaredConstructor(classOf[Class[_]], classOf[Int])
+      if (!methodHandlesLookupConstructor.isAccessible)
+        methodHandlesLookupConstructor.setAccessible(true)
+      method =>
+        // We create a MethodHandles.Lookup that is allowed to look up private methods
+        methodHandlesLookupConstructor.newInstance(method.getDeclaringClass, Integer.valueOf(MethodHandles.Lookup.PRIVATE))
+          // Now using unreflect special, we get the default method from the declaring class, rather than the proxy
+          .unreflectSpecial(method, method.getDeclaringClass)
+    } else {
+      val lookup = MethodHandles.lookup
+      val returnType = MethodType.methodType(classOf[Descriptor])
+      method =>
+        lookup.findSpecial(method.getDeclaringClass, method.getName, returnType, method.getDeclaringClass)
+    }
```

Comment:
Noob question. Why isn't:

```
      val lookup = MethodHandles.lookup
      val returnType = MethodType.methodType(classOf[Descriptor])
      method =>
        lookup.findSpecial(method.getDeclaringClass, method.getName, returnType, method.getDeclaringClass)
```

valid in Java 8?

Reply:
Ah, I've read a bit about https://openjdk.java.net/jeps/274.

Reply:
I'll add some inline links.